tests/kokkos/workspace_tests.cpp

```diff
@@ -98,6 +98,37 @@ static void unittest_workspace()
     REQUIRE(wsmd.m_reserve == 1);
     REQUIRE(wsmd.m_size == 17);
   }
+  {
+    // Test constructing the WSM using view data
+    size_t n_bytes = WorkspaceManager<double, Device>::get_total_bytes_needed(17, num_ws, policy);
+    size_t n_slots = n_bytes/sizeof(double);
+    view_1d<double> tmp_view("", n_slots);
+
+    // Calculate next available location
+    double* data_end = tmp_view.data();
+    data_end += tmp_view.size();
+
+    WorkspaceManager<double, Device> wsm(tmp_view.data(), 17, num_ws, policy);
```

Comment:
Maybe it's me, but I am getting a bit confused by the var names here. So, 17 is the size of each slot (as number of double's) in each workspace, right? Then num_ws sounds like 'num_workspaces', but it is the number of (17-doubles-long) slots needed in each workspace. And n_slots is not really the number of slots, but rather the number of doubles needed (which is (17+reserve)*num_workspaces*num_slots_per_ws), right?

Reply:
I think there are some things in the WSM stack that are not clear. E.g., `get_space_in_slot`, makes it sound like there are more than 1 space in a slot. But I _think_ the returned space _is a slot_.

Reply:
You are correct on all counts. I just pushed some new names.